src/FixedDataTable.js

```diff
@@ -318,16 +318,54 @@ class FixedDataTable extends React.Component {
 
     /**
      * Callback that is called when scrolling starts. The current horizontal and vertical scroll values,
-     * and the current first and last row indexes will be provided to the callback.
+     * the current first and last row indexes, and the current first and last column indexes
+     * will be provided to the callback.
+     *
+     * ```
+     * function(
+     *   scrollX, // the current horizontal scroll value
+     *   scrollY, // the current vertical scroll value
+     *   firstRowIndex: number, // the first visible row in the viewport
+     *   lastRowIndex: number, // the last visible row in the viewport
+     *   firstColumnIndex: number, // the first visible column in the viewport
+     *   lastColumnIndex: number, // the last visible column in the viewport
+     * )
+     * ```
      */
     onScrollStart: PropTypes.func,
 
     /**
      * Callback that is called when scrolling ends. The new horizontal and vertical scroll values,
-     * and the new first and last row indexes will be provided to the callback.
+     * the new first and last row indexes, and the new first and last column indexes
+     * will be provided to the callback.
+     *
+     * ```
+     * function(
+     *   scrollX, // the new horizontal scroll value
+     *   scrollY, // the new vertical scroll value
+     *   firstRowIndex: number, // the first visible row in the viewport
+     *   lastRowIndex: number, // the last visible row in the viewport
+     *   firstColumnIndex: number, // the first visible column in the viewport
+     *   lastColumnIndex: number, // the last visible column in the viewport
+     * )
+     * ```
      */
     onScrollEnd: PropTypes.func,
 
+    /**
+     * Callback that is called when the set of visible columns/rows inside the viewport changes.
+     *
+     * ```
+     * function({
+     *   firstRowIndex: number, // the first visible row in the viewport
+     *   lastRowIndex: number, // the last visible row in the viewport
+     *   firstColumnIndex: number, // the first visible column in the viewport
+     *   lastColumnIndex: number, // the last visible column in the viewport
+     * })
+     * ```
+     */
+    onViewportChange: PropTypes.func,
```

Comment:
Unlike `onScrollStart` and `onScrollEnd`, `onViewportChange` avoids passing a lengthy list of arguments by using an object as the parameter.

Unfortunately, we can't change `onScrollStart` and `onScrollEnd` to also use an object as param without breaking backward compatibility.